src/addons/FluxComponent.js

```diff
@@ -85,21 +85,25 @@ export default (React, PlainWrapperComponent) => {
         flux,
         ...extraProps } = this.props;
 
+      return extraProps;
+    }
+
+    getChildProps() {
       return assign(
-        extraProps,
+        this.getExtraProps(),
         this.state.storeState,
         this.state.actions
       );
     }
 
```

Comment:
A minor optimization, but I think you could get rid of `getChildProps()` here and just inline it inside `wrapChild()`. Then you don't have to call `this.getExtraProps()` twice.